src/transformers/trainer.py

```diff
@@ -272,6 +272,18 @@ def _get_fsdp_ckpt_kwargs():
         return {}
 
 
+def add_safe_globals():
+    # Starting from version 2.4 PyTorch introduces a stricter check for the objects which
+    # can be loaded with torch.load(). Starting from version 2.6 loading with weights_only=True
+    # requires allowlisting of such objects.
+    # See: https://github.com/pytorch/pytorch/pull/137602
+    # See: https://pytorch.org/docs/stable/notes/serialization.html#torch.serialization.add_safe_globals
+    if version.parse(torch.__version__) >= version.parse("2.4.0"):
+        torch.serialization.add_safe_globals(
+            [np.core.multiarray._reconstruct, np.ndarray, np.dtype, np.dtypes.UInt32DType]
```

Comment:
We could have a `SAFE_TRANSFORMERS_GLOBAL` with these no? this way people can easily update them? 
TBH I prefer the context manager but want to have the least duplication as possible!

Reply:
I found that calling `torch.serialization.add_safe_globals()` still works to add additional safe global staff. `SAFE_TRANSFORMERS_GLOBAL` can also be considered. Let me know if you see the need.